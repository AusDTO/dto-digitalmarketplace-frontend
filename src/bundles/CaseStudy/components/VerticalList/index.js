import VerticalList from './VerticalList'
export default VerticalList
