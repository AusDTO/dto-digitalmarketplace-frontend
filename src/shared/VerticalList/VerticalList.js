import React from 'react'
import PropTypes from 'prop-types';
const VerticalList = ({ items = [] }) => {
  if (!items.length) {
    return <span />
  }

  return (
    <ul className="list-vertical">
      {items.map((item, i) => (
        <li key={i}>
          <article>
            <h3 className="au-display-md"><a href={item.link}>{item.title}</a></h3>
            {item.meta ? <div className="meta" dangerouslySetInnerHTML={{ __html: item.meta }} /> : ''}
            {item.description ? <p>{item.description}</p> : ''}
          </article>
          {item.figure ? (
            <figure>
              <img src={item.figure.src} alt={item.figure.alt} />
            </figure>
          ) : ''}
        </li>
      ))}
  </ul>
  )
}

VerticalList.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape({
    link: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    meta: PropTypes.string,
    description: PropTypes.string,
    figure: PropTypes.shape({
      src: PropTypes.string.isRequired,
      alt: PropTypes.string.isRequired
    })
  }))
}


export default VerticalList
