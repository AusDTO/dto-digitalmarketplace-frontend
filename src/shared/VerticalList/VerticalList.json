{
  "requiredItems": [
    { "title": "Hello World", "link": "http://hello.world" }
  ],
  "items": [
    {
      "title": "Hello World",
      "link": "http://hello.world",
      "meta": "<time datetime=\"2016-05-08 00:00\">08 May 2016</time> <a href=\"#\" rel=\"author\">Author</a>",
      "description": "The Digital Transformation Office (DTO) was established as an executive agency in July 2015. Its mission is to lead the [â€¦]",
      "figure": {
        "src": "/img-placeholder.gif",
        "alt": "Default image for items without an image specified"
      }
    }
  ]
}
