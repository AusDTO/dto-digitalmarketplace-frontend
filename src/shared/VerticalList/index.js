import VerticalList from './VerticalList'
import testData from './VerticalList.json'
export { testData }
export default VerticalList
