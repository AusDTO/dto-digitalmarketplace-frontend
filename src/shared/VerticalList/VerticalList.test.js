import React from 'react'
import VerticalList from './VerticalList'
import renderer from 'react-test-renderer'

import testData from './VerticalList.json'

test('VerticalList with required attributes', () => {
  const component = renderer.create(
    <VerticalList items={testData.requiredItems} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('VerticalList with all attributes', () => {
  const component = renderer.create(
    <VerticalList items={testData.items} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('VerticalList with no attributes', () => {
  const component = renderer.create(
    <VerticalList />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
