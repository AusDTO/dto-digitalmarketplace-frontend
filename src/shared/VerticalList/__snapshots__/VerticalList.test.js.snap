exports[`VerticalList with all attributes 1`] = `
<ul
  className="list-vertical"
>
  <li>
    <article>
      <h3>
        <a
          href="http://hello.world"
        >
          Hello World
        </a>
      </h3>
      <div
        className="meta"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<time datetime="2016-05-08 00:00">08 May 2016</time> <a href="#" rel="author">Author</a>",
          }
        }
      />
      <p>
        The Digital Transformation Office (DTO) was established as an executive agency in July 2015. Its mission is to lead the […]
      </p>
    </article>
    <figure>
      <img
        alt="Default image for items without an image specified"
        src="/img-placeholder.gif"
      />
    </figure>
  </li>
</ul>
`;

exports[`VerticalList with no attributes 1`] = `<span />`;

exports[`VerticalList with required attributes 1`] = `
<ul
  className="list-vertical"
>
  <li>
    <article>
      <h3>
        <a
          href="http://hello.world"
        >
          Hello World
        </a>
      </h3>
      
      
    </article>
    
  </li>
</ul>
`;

exports[`test VerticalList with all attributes 1`] = `
<ul
  className="list-vertical">
  <li>
    <article>
      <h3>
        <a
          href="http://hello.world">
          Hello World
        </a>
      </h3>
      <div
        className="meta"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<time datetime=\"2016-05-08 00:00\">08 May 2016</time> <a href=\"#\" rel=\"author\">Author</a>",
          }
        } />
      <p>
        The Digital Transformation Office (DTO) was established as an executive agency in July 2015. Its mission is to lead the […]
      </p>
    </article>
    <figure>
      <img
        alt="Default image for items without an image specified"
        src="/img-placeholder.gif" />
    </figure>
  </li>
</ul>
`;

exports[`test VerticalList with no attributes 1`] = `<span />`;

exports[`test VerticalList with required attributes 1`] = `
<ul
  className="list-vertical">
  <li>
    <article>
      <h3>
        <a
          href="http://hello.world">
          Hello World
        </a>
      </h3>
      
      
    </article>
    
  </li>
</ul>
`;
